File: java/src/main/java/com/cams/challenge/UserRegistration.java
package com.cams.challenge;

import java.util.Map;
import java.util.List;

public class UserRegistration {

    public static class RegistrationResponse {
        public String status;
        public String userType;
        public String email;
        public List<String> errors;
        
        // Helper constructors you can use
        public static RegistrationResponse success(String type, String email) {
            RegistrationResponse res = new RegistrationResponse();
            res.status = "SUCCESS";
            res.userType = type;
            res.email = email;
            return res;
        }

        public static RegistrationResponse error(List<String> errors) {
            RegistrationResponse res = new RegistrationResponse();
            res.status = "ERROR";
            res.errors = errors;
            return res;
        }
    }

    /**
     * Validates user data, classifies user based on age, and returns response.
     *
     * @param rawData A map representing the raw user input data.
     *                Expected keys: 'firstName', 'lastName', 'email', 'phone', 'dateOfBirth'.
     * @return RegistrationResponse indicating SUCCESS or ERROR.
     */
    public RegistrationResponse registerUser(Map<String, String> rawData) {
        // TODO: Implement validation and classification logic here
        throw new UnsupportedOperationException("Not implemented yet");
    }
}

File: kotlin/src/main/kotlin/com/cams/challenge/UserRegistration.kt
package com.cams.challenge

data class RegistrationResponse(
    val status: String,
    val userType: String? = null,
    val email: String? = null,
    val errors: List<String>? = null
) {
    companion object {
        fun success(type: String, email: String) = RegistrationResponse("SUCCESS", type, email, null)
        fun error(errors: List<String>) = RegistrationResponse("ERROR", null, null, errors)
    }
}

/**
 * Validates user data, classifies user based on age, and returns response.
 *
 * @param rawData A map representing the raw user input data.
 *                Expected keys: 'firstName', 'lastName', 'email', 'phone', 'dateOfBirth'.
 * @return RegistrationResponse indicating SUCCESS or ERROR.
 */
fun registerUser(rawData: Map<String, String>): RegistrationResponse {
    // TODO: Implement validation and classification logic here
    throw NotImplementedError("Not implemented yet")
}
